components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/util/RuleBuilder.java

```diff
@@ -216,4 +238,91 @@ private void setValidationError(String message) {
+
+        if (!isValidValueType(fieldDefinition, value)) {
+            return value;
+        }
+
+        if (!isValidOptionsInputValue(fieldDefinition, rawValue)) {
+            return value;
+        }
+
+        try {
+            return resolveValue(fieldDefinition, rawValue);
+        } catch (RuleManagementClientException e) {
+            setValidationError(e.getMessage());
+            return value;
+        }
+    }
+
+    private Value resolveValue(FieldDefinition fieldDefinition,
+                               String rawValue) throws RuleManagementClientException {
+
+        org.wso2.carbon.identity.rule.metadata.model.Value.ValueType
+                fieldDefinitionValueType = fieldDefinition.getValue().getValueType();
+
+        switch (fieldDefinitionValueType) {
+            case STRING:
+                return new Value(Value.Type.STRING, rawValue);
+            case NUMBER:
+                return validateNumberValue(rawValue);
+            case BOOLEAN:
+                return validateBooleanValue(rawValue);
+            case REFERENCE:
+                return new Value(Value.Type.REFERENCE, rawValue);
+            default:
+                throw new RuleManagementClientException(
+                        "Unsupported value type: " + fieldDefinitionValueType + " for field: " +
+                                fieldDefinition.getField().getName());
+        }
+    }
+
+    private boolean isValidValueType(FieldDefinition fieldDefinition, Value value) {
+
+        org.wso2.carbon.identity.rule.metadata.model.Value.ValueType fieldDefinitionValueType =
+                fieldDefinition.getValue().getValueType();
+        Value.Type valueType = value.getType();
+
+        if (valueType != Value.Type.RAW && !valueType.name().equals(fieldDefinitionValueType.name())) {
+            setValidationError(
+                    "Value type " + valueType + " is not supported for field " + fieldDefinition.getField().getName());
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isValidOptionsInputValue(FieldDefinition fieldDefinition, String fieldValue) {
+
+        if (fieldDefinition.getValue() instanceof OptionsInputValue &&
+                ((OptionsInputValue) fieldDefinition.getValue()).getValues().stream()
+                        .noneMatch(optionsValue -> optionsValue.getName().equals(fieldValue))) {
+            setValidationError(
+                    "Value " + fieldValue + " is not supported for field " + fieldDefinition.getField().getName());
+            return false;
+        }
+
+        return true;
+    }
+
+    private Value validateNumberValue(String rawValue) throws RuleManagementClientException {
+
+        try {
+            Double.parseDouble(rawValue);
+            return new Value(Value.Type.NUMBER, rawValue);
+        } catch (NumberFormatException e) {
+            throw new RuleManagementClientException("Value " + rawValue + " is not a valid NUMBER.");
```

Comment:
Is there a reason throwing RuleManagementClientException instead of setting setValidationError() as other places.

Reply:
Exception is centrally handled and setValidationError is invoked to reduce unnecessary clutter. if 'setValidationError()' is used the method should return at that instance.

Refer https://github.com/wso2/carbon-identity-framework/pull/6265/files#diff-2b1775cf9e6e88f9f97b94276b6b9891609926be63f80c649bc953ee5258b7b6R254-R282